frontend/src/views/SubmissionView.vue

```diff
@@ -42,28 +59,88 @@ async function onSubmit(data: any) {
 
 onMounted(async () => {
   submission.value = (await chefsService.getSubmission(props.formId, props.submissionId)).data;
+  documents.value = (await documentService.listDocuments(props.submissionId)).data;
 });
-
-const onUpload = async (file: File) => {
-  await documentService.createDocument(file, props.submissionId, getConfig.value.coms.bucketId);
-};
 </script>
 
 <template>
   <h1>Activity submission</h1>
 
-  <Button
-    class="mb-3"
-    @click="editable = !editable"
-  >
-    Edit
-  </Button>
-
-  <SubmissionForm
-    v-if="submission"
-    :editable="editable"
-    :submission="submission"
-    @cancel="onCancel"
-    @submit="onSubmit"
-  />
+  <TabView>
+    <TabPanel header="Info">
+      <Button
+        class="mb-3"
+        @click="editable = !editable"
+      >
+        Edit
+      </Button>
+
+      <SubmissionForm
+        v-if="submission"
+        :editable="editable"
+        :submission="submission"
+        @cancel="onCancel"
+        @submit="onSubmit"
+      />
+    </TabPanel>
+    <TabPanel header="Files">
+      <div class="grid border-1 surface-border border-round text-center">
+        <div class="col-2">
+          <FileUpload
+            :submission-id="props.submissionId"
+            @update:model-value="(e: Document) => documents.push(e)"
+          />
+        </div>
+        <div class="col-10 p-0">
+          <Carousel
```

Comment:
youre probably going to want this upload ui in a separate component eventually.
I'm still questioning the need for a carousel at this stage and the complications that come with it, but good work figuring it out. The primevue component does look pretty good. Although I think we could adjust the size, spacing and default thumbnail lol.